pkg/git/libgit2/checkout_test.go

```diff
@@ -25,40 +25,70 @@ import (
 	"testing"
 	"time"
 
+	"github.com/fluxcd/pkg/gittestserver"
+	feathelper "github.com/fluxcd/pkg/runtime/features"
+	"github.com/fluxcd/source-controller/internal/features"
 	"github.com/fluxcd/source-controller/pkg/git"
+	"github.com/fluxcd/source-controller/pkg/git/libgit2/managed"
+	"github.com/go-logr/logr"
 	git2go "github.com/libgit2/git2go/v33"
 	. "github.com/onsi/gomega"
 )
 
-func TestCheckoutBranch_checkoutUnmanaged(t *testing.T) {
-	repo, err := initBareRepo(t)
+func TestMain(m *testing.M) {
+	code := m.Run()
+	if code != 0 {
+		os.Exit(code)
+	}
+
+	enableManagedTransport()
+	// execute test against managed transport
+	code = m.Run()
+	os.Exit(code)
```

Comment:
I'm not sure how much we gain by using TestMain to enable managed transport.
Previously, the managed transport specific tests had initialization code, and now, the non-managed transport tests have the feature check and skip code. Doesn't seem to be any better. And in addition to that, some tests are now run twice that don't have anything specific to the transport.
I think it was fine how it was before.

Reply:
Addressed by moving the managed transport related tests into a separate file which is run at the very end.